VoodooI2CHID/VoodooI2CHIDDevice.cpp

```diff
@@ -411,15 +425,11 @@ IOReturn VoodooI2CHIDDevice::setPowerState(unsigned long whichState, IOService*
         if (!awake) {
             awake = true;
 
-            setHIDPowerState(kVoodooI2CStateOn);
-
-            VoodooI2CHIDDeviceCommand command;
-            command.c.reg = hid_descriptor.wCommandRegister;
-            command.c.opcode = 0x01;
-            command.c.report_type_id = 0;
-
-            api->writeI2C(command.data, 4);
-            IOSleep(10);
+            if (quirks & I2C_HID_QUIRK_RESET_ON_RESUME) {
+                resetHIDDevice();
+            } else {
+                setHIDPowerState(kVoodooI2CStateOn);
+            }
```

Comment:
Is it safe to remove the `IOSleep` here?

Reply:
`setHIDPowerState` sleeps for `100ms` after setting it.
`resetHIDDevice` calls `setHIDPowerState` inside, then either waits for an IRQ (on almost all devices) or waits `100ms` after the reset.

Reply:
Makes sense, go for the merge!